src/generators.rs

```diff
@@ -34,6 +34,15 @@ where
     left.zip(right)
 }
 
+#[inline]
+fn get_num_nodes(num_nodes: &Option<usize>, weights: &Option<Vec<PyObject>>) -> usize {
+    if weights.is_some() {
+        weights.as_ref().unwrap().len()
+    } else {
+        num_nodes.unwrap()
+    }
+}
```

Comment:
We might want to ensure the output is non-zero and raise a `ValueError` if not, since later we might substract 1.

Reply:
Fixed in: https://github.com/Qiskit/retworkx/pull/625/commits/4d4c9ed740072d7278c4392f7cba16e78ca56bcf I opted to just explicitly return an empty graph if node_len == 0. This is what used to happen if it was a 0 len input for most of them before